new-packages/ts-project/src/state.ts

```diff
@@ -30,6 +29,40 @@ const createActionBlock = ({
     },
   };
 };
+export function createActorBlock({
+  sourceId,
+  parentId,
+  actorId,
+}: {
+  sourceId: string;
+  parentId: string;
+  actorId: string;
+}): ActorBlock {
+  const blockId = uniqueId();
+  return {
+    blockType: 'actor',
+    uniqueId: blockId,
+    parentId,
+    sourceId,
+    properties: {
+      src: sourceId,
+      id: actorId,
+    },
+  };
+}
+
+function getActorId(
```

Comment:
overall this isn't a bad idea but I struggle to see why we are doing this here now if we didn't do this for anything else? 🤔 What makes this special?

Reply:
Do you mean the function itself or what we're doing to the actor id?

Reply:
> or what we're doing to the actor id?

This. What makes actor id special? I don't recall a similar pattern used anywhere so far. Like, we didn't do anything like this to `history: someVariable`.

As I mentioned, I don't think this is bad - I just wonder if there is any specific reason/line of thought that made you to do this specifically here.

Reply:
I checked Studio and we don't update this value in any special way. The only thing that Studio cares about is that if user didn't provide any `id` to the actor, the id that's extracted must be in a special format so Studio can render it as `anonymous`.

Reply:
so this could be simplified to smth like this, right?
```ts
`inline:${uniqueId()}`
```

Reply:
It can be anything that doesn't include `:invocation`, apparently. Below is the Studio code responsible for this check:
```ts
export const isUserOverrideId = (
  id: string | undefined,
): id is SerialisableInvocationId => {
  if (!id) return false;
  return !id.includes(':invocation');
};
```
But we can be consistent and use the same format we have already.